app/models/assignment.rb

```diff
@@ -1247,29 +1247,97 @@ def to_json(options = {})
     self.assignment_properties.attributes.merge(self.attributes).symbolize_keys.to_json(options)
   end
 
+  # Returns an assignment's relevant properties for uploading/downloading an assignment's configuration as a hash
+  def assignment_properties_config
+    # Data to avoid including
+    exclude = %w[id created_at updated_at outstanding_remark_request_count repository_folder]
+    should_reject = ->(attr) { attr.end_with?('_id', '_created_at', '_updated_at') }
```

Comment:
I'm not sure what the `_created_at` and `_updated_at` suffixes are checking for?

Reply:
The `starter_file_updated_at attribute` in the assignment_properties model has to be rejected (adds unnecessary information that will be overwritten after uploading). 
The `_created_at` was added for future-proofing. If an attribute similar to `created_at` is made, it likely won't be necessary for this feature and can therefore be ignored.

Reply:
I requested this change https://github.com/MarkUsProject/Markus/pull/5616#discussion_r761197563